### DISCUSSION TITLE
Mojo::DOM vs tag named link

### DISCUSSION THREAD

**Discussion by daleif** [mojolicious/mojo#2042] [General]
I might be missing something from the documentation, but why does

```
say Mojo::DOM->new('<link>test</link>')->xml(1)->to_string;
say Mojo::DOM->new('<link>test</link>')->xml(0)->to_string;
```

give 

```
<link />test
<link>test
```

and not

```
<link>test</link>
```

in both cases?

I was playing with some XML/RSS data and was wondering why I got a strange output when printing to terminal.

It seems to be triggered by the name of the tag as the tag `test` behaves as expected.

**Comment by daleif** [comment]
Ahh, right, the `->xml(1)` comes too late. One ought to use

```
say Mojo::DOM->new->xml(1)->parse('<link>test</link>')->to_string;
```

**Reply by mauke** [reply]
Right. In HTML, the [link element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) is "void", i.e. it cannot have any content (and the start tag automatically closes itself).

**Reply by daleif** [reply]
@mauke where is that listed in the sources? I think I was looking for it at one point

**Reply by mauke** [reply]
https://github.com/mojolicious/mojo/blob/a46fbb7ad5cb62b3404c7515bb20d827c9472102/lib/Mojo/DOM/HTML.pm#L78
https://github.com/mojolicious/mojo/blob/a46fbb7ad5cb62b3404c7515bb20d827c9472102/lib/Mojo/DOM/HTML.pm#L143
